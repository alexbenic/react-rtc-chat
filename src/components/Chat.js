import React from "react";
import MessageList from "./MessageList";
import MessageForm from "./MessageForm";
import SideNav from "./SideNav";

const Chat = ({ current, users, messages, stats, onSubmit }) => {
  return (
    <div className="flex flex-column vh-100">
      <div className="flex flex-auto">
        <SideNav current={current} users={users} stats={stats} />
        <section className="pa2 w-80 flex flex-column">
          <MessageList current={current} messages={messages} />
          <MessageForm onSubmit={onSubmit} />
        </section>
      </div>
    </div>
  );
};

export default Chat;
