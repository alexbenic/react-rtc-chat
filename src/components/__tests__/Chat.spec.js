import React from "react";
import each from "jest-each";
import { render, cleanup, fireEvent } from "react-testing-library";
import randombytes from "randombytes";
import Chat from "../Chat";

beforeEach(cleanup);

const user = name => {
  return {
    id: randombytes(8).toString("hex"),
    nick: name,
    stream: {},
    app: 1
  };
};

const message = text => {
  return {
    id: randombytes(8).toString("hex"),
    sender: user("zvrk"),
    message: text
  };
};

// prettier-ignore
each([
  [ "zvrk", [], [], {angular: 0, react: 1} ],
  [ "zvrk", [], [message('anyone here?')], {angular: 0, react: 1} ],
  [ "zook" ,[user('zvrk')], [message('yo'), message('hi')], { angular: 1, react: 1 } ],
])
.test("should render chat for %s with user list: %j and message list: %j", ( nick, users, messages, stats ) => {

  const handleSubmit = jest.fn();
  const { container, getByPlaceholderText } = render(
    <Chat current={user(nick)} users={users} messages={messages} stats={stats} onSubmit={handleSubmit} />
  );

  const input = getByPlaceholderText("Type a message here then hit ENTER");
  const form = container.querySelector('form')
  input.value = 'hello';

  fireEvent.change(input)
  fireEvent(form, new Event("submit"));

  expect(handleSubmit).toHaveBeenCalledTimes(1);
  expect(handleSubmit).toHaveBeenCalledWith('hello');
});

test("chat snapshot", () => {
  const handleSubmit = jest.fn();
  const current = user("zvrk");
  const users = [user("zoom"), user("maci")];
  const messages = [message("hello")];

  const { container } = render(
    <Chat
      current={current}
      users={users}
      messages={messages}
      stats={{ angular: 1, react: 2 }}
      onSubmit={handleSubmit}
    />
  );

  expect(container).toMatchSnapshot();
});
