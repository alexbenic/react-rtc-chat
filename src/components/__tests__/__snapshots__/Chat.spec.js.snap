// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat snapshot 1`] = `
<div>
  <div
    class="flex flex-column vh-100"
  >
    <div
      class="flex flex-auto"
    >
      <aside
        class="w-20 pa3 flex flex-column bg-sidebar"
      >
        <img
          class="db w-100 h4 center ofc"
          src="react_logo.png"
          alt="react logo"
        />
        <span
          data-testid="current-nick"
          title="zvrk"
          class="pa2 bb bw-1 tc truncate"
        >
          zvrk
        </span>
        <div
          class="flex-auto mt2"
        >
          <li
            class="flex items-center mt2 mb2 pb1 pt1"
          >
            <div
              class="br-100 w1 h1 mr3 bg-green"
            />
            zoom
          </li>
          <li
            class="flex items-center mt2 mb2 pb1 pt1"
          >
            <div
              class="br-100 w1 h1 mr3 bg-green"
            />
            maci
          </li>
        </div>
        <div
          class="flex flex-column flex-row-ns flex-none-ns h4 h3-ns ma2"
        >
          <div
            class="flex flex-auto flex-column justify-center items-center pa1"
          >
            <img
              src="angular_logo.png"
              class="db w3 h3"
              alt="angular logo"
            />
            <span
              class="tc"
            >
              1
            </span>
          </div>
          <div
            class="flex flex-auto flex-column justify-center items-center p1"
          >
            <img
              src="react_logo.png"
              class="db w3 h3"
              alt="react logo"
            />
            <span>
              2
            </span>
          </div>
        </div>
      </aside>
      <section
        class="pa2 w-80 flex flex-column"
      >
        <div
          class="flex-auto"
        >
          <ul
            class="list pl2"
          >
            <li
              class="mv1 l-list pt3 bb bw1 b--light-gray"
            >
              <span
                class="b mv2 b2 ba bw1 b--light-gray pv1 ph2 br3"
              >
                zvrk
              </span>
              <p
                class="l-list f5 ph2"
              >
                hello
              </p>
            </li>
          </ul>
        </div>
        <div
          class="pt4 pb4 pl2 ma2 bt bw1 b--black-40"
        >
          <form
            class="flex"
          >
            <input
              type="text"
              class="flex-auto f5 fw5 bg-none b--none outline-0"
              placeholder="Type a message here then hit ENTER"
              value=""
            />
          </form>
        </div>
      </section>
    </div>
  </div>
</div>
`;
